lib/registry.js

```diff
@@ -16,7 +17,9 @@ export function define(id, spec) {
   if (get(id)) {
     console.warn(`Overwriting spec ${getName(id)}`)
   }
-  spec.name = getName(id)
+  if (!spec.name && typeof spec !== "symbol") {
```

Comment:
Ah I didn't know strings could be used for ids. Perhaps they should be converted to Symbols internally? That would save doing multiple type checks on id.

I'll check the predicates; perhaps use lodash?